Fix ABI breakage in 0.8.1

KitsuneRal opened the pull request:
See the commit. Closes #702.

KitsuneRal committed:
Revert constifying Room::userIdsAtEvent()

To avoid ABI breakage within 0.8.x. Fixes #702.

nvrWhere commented:
So I understand it broke the ABI but we're now reliant on this in Neochat. It would cause it not to compile.

KitsuneRal commented:
:( Okay, guess we'll have to have both signatures side by side until 0.9 then. Fortunately this function is not virtual.

KitsuneRal committed:
Have const and non-const signatures side by side

To satisfy Neochat that already ended up depending on the new (const)
signature.
